rt/rs/description-swagger-ui/src/main/java/org/apache/cxf/jaxrs/swagger/ui/SwaggerUiResourceLocator.java

```diff
@@ -50,8 +50,14 @@ public URL locate(String resourcePath) throws MalformedURLException {
         if (resourcePath.startsWith("/")) {
             resourcePath = resourcePath.substring(1);
         }
+        URL ret;
 
-        return URI.create(swaggerUiRoot + resourcePath).toURL();
+        try {
+            ret = URI.create(swaggerUiRoot + resourcePath).toURL();
+        } catch (Exception ex) {
```

Comment:
The `URI.create` may thrown only `IllegalArgumentException` exception, could you please use it instead of generic `Exception`? Thank you.

Reply:
I can do that, but the way that the code is currently implemented, the only caller is looking for a MalformedURLException if there is any problem in this method, and the request chain gets nuked if any other exception is thrown which is kinda catastrophic.  Was thinking that any exception from this method should raise the MalformedURLException so the caller just returns false. Future proofing.  If you still want the specific exception caught I can edit and resubmit.

Reply:
Thanks for the response, it is fine to raise `MalformedURLException` as you did, the `URI.create` does not do anything beside parsing and throwing `IllegalArgumentException`:
```
        try {
            return URI.create(swaggerUiRoot + resourcePath).toURL();
        } catch (IllegalArgumentException ex) {
            throw new MalformedURLException(ex.getMessage(), ex);
        }
```
Thank you.

Reply:
To avoid additional knowledge about internal RuntimeException rising a regular constructor with checked URISyntaxException can be used:
```
        try {
            return new URI(swaggerUiRoot + resourcePath).toURL();
        } catch (URISyntaxException ex) {
            throw new MalformedURLException(ex.getMessage());
        }
```